sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/faultinjection/FaultInjectBadSessionTokenTests.java

```diff
@@ -0,0 +1,206 @@
+        this.cosmosAsyncContainer = getSharedMultiPartitionCosmosContainerWithIdAsPartitionKey(cosmosAsyncClient);
+        this.writeRegionMap = this.getRegionMap(databaseAccount, true);
+    }
+
+    @Test(groups = {"multi-master"}, timeOut = 60000)
+    public void readSessionUnavailable_regionWide_test() {
+
+        List<String> preferredLocations = this.writeRegionMap.keySet().stream().collect(Collectors.toList());
+
+        assertThat(preferredLocations).isNotNull();
+        assertThat(preferredLocations.size()).isEqualTo(2);
+
+        CosmosAsyncClient clientWithPreferredRegions = null;
+
+        clientWithPreferredRegions = new CosmosClientBuilder()
+                .endpoint(TestConfigurations.HOST)
+                .key(TestConfigurations.MASTER_KEY)
+                .contentResponseOnWriteEnabled(true)
+                .consistencyLevel(BridgeInternal.getContextClient(this.cosmosAsyncClient).getConsistencyLevel())
+                .preferredRegions(preferredLocations)
+                .directMode()
+                .buildAsyncClient();
+
+        CosmosAsyncContainer containerForClientWithPreferredRegions = clientWithPreferredRegions
+                .getDatabase(this.cosmosAsyncContainer.getDatabase().getId())
+                .getContainer(this.cosmosAsyncContainer.getId());
+
+        TestItem createdItem = TestItem.createNewItem();
+        containerForClientWithPreferredRegions.createItem(createdItem).block();
+
+        FaultInjectionRuleBuilder badSessionTokenRuleBuilder = new FaultInjectionRuleBuilder("serverErrorRule-bad-session-token-" + UUID.randomUUID());
+
+        FaultInjectionCondition faultInjectionConditionForReadsInPrimaryRegion = new FaultInjectionConditionBuilder()
+                .operationType(FaultInjectionOperationType.READ_ITEM)
+                .connectionType(FaultInjectionConnectionType.DIRECT)
+                .region(preferredLocations.get(0))
+                .build();
+
+        FaultInjectionServerErrorResult badSessionTokenServerErrorResult = FaultInjectionResultBuilders
+                .getResultBuilder(FaultInjectionServerErrorType.BAD_SESSION_TOKEN)
+                .build();
+
+        FaultInjectionRule badSessionTokenRule = badSessionTokenRuleBuilder
+                .condition(faultInjectionConditionForReadsInPrimaryRegion)
+                .result(badSessionTokenServerErrorResult)
+                .duration(Duration.ofMinutes(10))
+                .build();
+
+        CosmosFaultInjectionHelper
+                .configureFaultInjectionRules(containerForClientWithPreferredRegions, Arrays.asList(badSessionTokenRule))
+                .block();
+
+        CosmosItemResponse<TestItem> itemResponse = containerForClientWithPreferredRegions.readItem(
+                createdItem.getId(),
+                new PartitionKey(createdItem.getId()),
+                TestItem.class).block();
+
+        assertThat(itemResponse.getStatusCode()).isEqualTo(HttpConstants.StatusCodes.OK);
```

Comment:
maybe also testing the duration of the operation, like by default it will at least take 5s